src/RealtimeClient.ts

```diff
@@ -96,6 +97,7 @@ export default class RealtimeClient {
   heartbeatIntervalMs: number = 25000
   heartbeatTimer: ReturnType<typeof setInterval> | undefined = undefined
   pendingHeartbeatRef: string | null = null
+  heartbeatCallback: Function = noop
```

Comment:
wondering if instead we should go with a `onHeartbeat` method instead of assigning the callback to a variable.

Reply:
how would it look like usage wise? and would it be implemented similarly but it would override the value?

Reply:
If we choose to use the `onHeartbeat` method, it should be able to handle multiple callbacks.

I’m leaning towards `onHeartbeat` because it better fits what we already have in the library, like `onSystem`. I think users expect to listen for any event using a `on` method.

As for the implementation, we could add a new property to the `stateChangeCallbacks` for the heartbeat, using the structure we already have.

Reply:
@filipecabaco another line of thought.

Whats is the idea behind exposing the heartbeat callback? Is it to offer ways of the user to debug the client?

If debugging is the reason, we could even enhance this idea and provide a `HeartbeatStatus` type, something like:

```ts
type HeartbeatStatus = 'sent' | 'received' | 'timed out'

onHeartbeat<T extends { [key: string]: any }>(
    callback: (msg?: T, status: HeartbeatStatus) => void
)
```

Just throwing a few ideas, lets make sure this improves the lib, and not just add something that won't be used.

Reply:
the idea is for users to react to realtime failure. e.g. if nothing arrives to this callback within the timer then we fail... so even the current implementation is lacking 😓 

taking that into consideration how do you think we should approach it?

Reply:
applied the changes suggested and the callback calls on heartbeat status